granulate_utils/metadata/databricks_client.py

```diff
@@ -43,46 +50,80 @@ def get_webui_address() -> Optional[str]:
 
     def get_job_name(self) -> Optional[str]:
         # Retry in case of a connection error, as the metrics server might not be up yet.
-        for i in range(MAX_RETRIES):
-            time.sleep(10)
+        retries = 0
+        while retries < MAX_RETRIES:
+            retries += 1
             try:
-                return self._get_job_name_impl()
-            except Exception:
-                self.logger.exception("Got Exception while collecting Databricks job name.")
+                if cluster_metadata := self._cluster_all_tags_metadata():
+                    # Got the job name, no need to retry.
+                    name = self._get_name_from_metadata(cluster_metadata)
+                    if name:
+                        self.logger.debug("Found name in metadata", job_name=name, cluster_metadata=cluster_metadata)
+                        return name
+                    else:
+                        self.logger.debug("Failed to extract name from metadata", cluster_metadata=cluster_metadata)
+                        return None
+                else:
+                    # No job name yet, retry.
+                    time.sleep(RETRY_INTERVAL)
+            except DatabricksJobNameDiscoverException as e:
+                self.logger.exception("Failed to get Databricks job name", exception=e)
```

Comment:
Any reason you pass `exception=e` in `logger.exception` calls? `logging` already logs the exception when you use `logginer.exception`.

Reply:
5326fee